RNVectorIcons.podspec

```diff
@@ -1,36 +1,20 @@
 require 'json'
-version = JSON.parse(File.read('package.json'))["version"]
-
-folly_compiler_flags = '-DFOLLY_NO_CONFIG -DFOLLY_MOBILE=1 -DFOLLY_USE_LIBCPP=1 -Wno-comma -Wno-shorten-64-to-32'
+package = JSON.parse(File.read(File.join(__dir__, "package.json")))
 
 Pod::Spec.new do |s|
-
   s.name           = "RNVectorIcons"
-  s.version        = version
-  s.summary        = "Customizable Icons for React Native with support for NavBar/TabBar, image source and full styling."
-  s.homepage       = "https://github.com/oblador/react-native-vector-icons"
-  s.license        = "MIT"
-  s.author         = { "Joel Arvidsson" => "joel@oblador.se" }
-  s.platforms      = { :ios => "9.0", :tvos => "9.0" }
-  s.source         = { :git => "https://github.com/oblador/react-native-vector-icons.git", :tag => "v#{s.version}" }
-  s.source_files   = 'RNVectorIconsManager/**/*.{h,m}'
+  s.version        = package["version"]
+  s.summary        = package["description"]
+  s.description    = package["description"]
+  s.homepage       = package["homepage"]
+  s.license        = package["license"]
+  s.author         = { package["author"]["name"] => package["author"]["email"] }
+  s.platforms      = { :ios => "12.0", :tvos => "9.0" }
+  s.source         = { :git => package["repository"]["url"], :tag => "v#{s.version}" }
+
+  s.source_files   = 'RNVectorIconsManager/**/*.{h,m,mm,swift}'
   s.resources      = "Fonts/*.ttf"
   s.preserve_paths = "**/*.js"
-  s.dependency 'React-Core'
-
-  # This guard prevent to install the dependencies when we run `pod install` in the old architecture.
-  if ENV['RCT_NEW_ARCH_ENABLED'] == '1' then
-      s.compiler_flags = folly_compiler_flags + " -DRCT_NEW_ARCH_ENABLED=1"
-      s.pod_target_xcconfig    = {
-          "HEADER_SEARCH_PATHS" => "\"$(PODS_ROOT)/boost\"",
-          "CLANG_CXX_LANGUAGE_STANDARD" => "c++17"
-      }
-
-      s.dependency "React-Codegen"
-      s.dependency "RCT-Folly"
-      s.dependency "RCTRequired"
-      s.dependency "RCTTypeSafety"
-      s.dependency "ReactCommon/turbomodule/core"
-  end
-
+  # React Native Core dependency
+  install_modules_dependencies(s)
```

Comment:
This breaks b/w compatibility with any version before 0.71 (even if they are not using fabric). I'm fine with breaking fabric itself for 0.70 users because they are few and very likely on latest version anyways. Could we perhaps change it to call it only if the function exists?

Reply:
I'm thinking something simpler like this: 

```suggestion
  if defined? install_modules_dependencies
    install_modules_dependencies(s)
  end
```

WDYT?

Reply:
I think this is simple and avoids having to care/know about RN specifics such as the `RCT_NEW_ARCH_ENABLED` env variable etc.